pkg/purge/purge.go

```diff
@@ -0,0 +1,221 @@
+package purge
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/fatih/color"
+	"github.com/glasskube/glasskube/internal/clientutils"
+	"github.com/glasskube/glasskube/internal/config"
+	"github.com/glasskube/glasskube/internal/httperror"
+	"github.com/glasskube/glasskube/internal/releaseinfo"
+	"github.com/glasskube/glasskube/internal/telemetry"
+	"github.com/glasskube/glasskube/internal/telemetry/annotations"
+	"github.com/glasskube/glasskube/pkg/bootstrap"
+	"go.uber.org/multierr"
+	"k8s.io/apimachinery/pkg/api/errors"
+
+	"github.com/schollz/progressbar/v3"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/restmapper"
+)
+
+type PurgeClient struct {
+	clientConfig *rest.Config
+	mapper       meta.RESTMapper
+	client       dynamic.Interface
+}
+
+type PurgeOptions struct {
+	Type                    bootstrap.BootstrapType
+	Url                     string
+	Latest                  bool
+	CreateDefaultRepository bool
+	DisableTelemetry        bool
+	Force                   bool
+}
+
+func DefaultPurgeOptions() PurgeOptions {
+	return PurgeOptions{Type: bootstrap.BootstrapTypeAio, Latest: config.IsDevBuild(), CreateDefaultRepository: true}
+}
+
+func NewPurgeClient(config *rest.Config) *PurgeClient {
+	return &PurgeClient{clientConfig: config}
+}
+
+func (c *PurgeClient) initRestMapper() error {
+	if discoveryClient, err := discovery.NewDiscoveryClientForConfig(c.clientConfig); err != nil {
+		return err
+	} else if groupResources, err := restmapper.GetAPIGroupResources(discoveryClient); err != nil {
+		return err
+	} else {
+		c.mapper = restmapper.NewDiscoveryRESTMapper(groupResources)
+		return nil
+	}
+}
+
+func (c *PurgeClient) Purge(ctx context.Context, options PurgeOptions) error {
+	if err := c.initRestMapper(); err != nil {
+		return err
+	}
+
+	if client, err := dynamic.NewForConfig(c.clientConfig); err != nil {
+		return err
+	} else {
+		c.client = client
+	}
+
+	fmt.Println("Starting purge process")
+	start := time.Now()
+
+	if options.Url == "" {
+		version := config.Version
+		if options.Latest {
+			if releaseInfo, err := releaseinfo.FetchLatestRelease(); err != nil {
+				if httperror.Is(err, http.StatusServiceUnavailable) || httperror.IsTimeoutError(err) {
+					telemetry.BootstrapFailure(time.Since(start))
+					return fmt.Errorf("network connectivity error, check your network: %w", err)
+				}
+				telemetry.BootstrapFailure(time.Since(start))
+				return fmt.Errorf("could not determine latest version: %w", err)
+			} else {
+				version = releaseInfo.Version
+			}
+		}
+		options.Url = fmt.Sprintf("https://github.com/glasskube/glasskube/releases/download/v%v/manifest-%v.yaml",
+			version, options.Type)
+	}
```

Comment:
I think for now we should just set the URL to the latest version and "slim" variant in order not to delete the flux components (this was mentioned in an issue comment).

We can introduce flags/options to change this behavour later.